### ISSUE TITLE
reloading not working with webcam 

### ISSUE THREAD

**Issue by djmvt** [Oitzu/pimatic-iframe#8]
When creating an Iframe with a webcam snapshot link auto reload seems to do nothing.
Tried the reload variable but that doesn't change the picture.

When opening the link in a browser it refreshes the picture everytime I reload it.

Any clues?

**Comment by Oitzu** [comment]
Do you have an example link what's the url to the Webcam image looks like?

**Comment by Oitzu** [comment]
@djmvt i've seen you solved this in https://forum.pimatic.org/topic/1439/iframe-device-won-t-refresh-picture-from-ip-camera/2
Could you please post your solution? I will then add it to the documentation.

**Comment by djmvt** [comment]
It is a somewhat dirty trick but it is working (not my idea, user KoffieNL 
pointed me in the right direction)

What I did is using the system variable for cpu usage as a "fake" url by 
appending it to the camera url like so: 
"http://camera_url:port/webcapture.jpg?command=snap&channel=0[? 
$syssensor.cpu]"

The part between the [ brackets] is what I added to the url.
This way the url changes everytime it is loaded so the image gets refreshed.

I haven't tested this in the device settings of iframe because I have no 
use for constant updates of the iframe. Using this in a rule now to let the 
image get updated everytime the rule is started.

Perhaps a change in the code that auto appends a random string to the url 
could be an option for a permanent solution?

Op 22 februari 2016 1:40:24 p.m. schreef Oitzu notifications@github.com:

**Comment by Oitzu** [comment]
Oh, so it's a iframe caching problem. Yeah, maybe something like appending ?[Random Number] to the url would be a solution.

**Comment by mwittig** [comment]
> Oh, so it's a iframe caching problem.

I don't think it is a general caching problem. I have tested with loading a jpeg using a background shell script replacing the content of the jpeg file every two seconds. The iframe content gets updated on every reload cycle (set to 10 seconds) with Chrome and Firefox. So, there must be something more specific about the webcamera case.

What does the camera URL look like? Does the image file on disc changes including time stamp?

Here is my test case: I am using pimatic-filebrowser to access the file. The URL is http://raspi2/files/pic.jpg?command=snap, I am using the following script to replace the content:

```
#!/bin/bash

loop() {
        for i in "1.jpg" "2.jpg" "3.jpg"
        do
                cat $i > pic.jpg
                sleep 2
        done
}
while true; do loop; done
```

I also reviewed the code which is actually [performing the reload](https://github.com/Oitzu/pimatic-iframe/blob/master/app/iframe-page.coffee#L59-L64) and it is already using the dirty trick to enforce a reload as decribed in http://stackoverflow.com/a/4062084/4816693

**Comment by mwittig** [comment]
Btw, @djmvt, which value do you use for "reload". Note, the value is given in seconds. So, `"reload": 10` will trigger a reload every 10 seconds.

**Comment by djmvt** [comment]
http://ipadress:port/webcapture.jpg?command=snap&channel=0 
This is the url used by my camera.
I tried the reload function (in seconds) but it never updated the picture.
Could be it is just a problem with my cam ofcourse, it is working with the random variable added to the url (added ?$syssensor.cpu)

**Comment by Oitzu** [comment]
@djmvt could you confirm that this issue is resolved with version 0.2.0?
